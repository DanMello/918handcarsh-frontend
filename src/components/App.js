import React, { Component } from 'react';
import styles from 'styles/App.css';

export default class App extends Component {

render() {
  return <div id="test" className={styles.welcome}>Welcome to your app.</div>
  };
};