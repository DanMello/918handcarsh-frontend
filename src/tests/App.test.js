import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import App from '../components/App';
import renderer from 'react-test-renderer';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it('Renders without crashing', () => {
  ReactDOM.render(<App />, container);
});

it('Renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders Welcome to your app.', () => {
  act(() => {
    ReactDOM.render(<App />, container);
  });
  expect(container.querySelector('#test').innerHTML).toBe('Welcome to your app.')
});