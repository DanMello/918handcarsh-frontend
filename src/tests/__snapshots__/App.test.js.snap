// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="welcome"
  id="test"
>
  Welcome to your app.
</div>
`;
